packages/bun-vscode/src/features/tests/bun-test-controller.ts

```
@@ -0,0 +1,1424 @@
+  }
+
+  private async discoverInitialTests(): Promise<void> {
+    try {
+      const tests = await this.findTestFiles();
+      this.createFileTestItems(tests);
+    } catch (error) {
+      // Silent error handling
+    }
+  }
+
+  private async findTestFiles(): Promise<vscode.Uri[]> {
+    return await this.findTestFilesWithGitignore();
+  }
+
+  private customFilePattern(): string {
+    return vscode.workspace.getConfiguration("bun.test").get("filePattern", DEFAULT_TEST_PATTERN);
+  }
+
+  private async findTestFilesWithGitignore(): Promise<vscode.Uri[]> {
+    const ignoreGlobs = await this.buildIgnoreGlobs();
+    const tests = await vscode.workspace.findFiles(this.customFilePattern(), "node_modules");
+
+    return tests.filter(test => {
+      const normalizedTestPath = test.fsPath.replace(/\\/g, "/");
+      return !ignoreGlobs.some(glob => {
+        const normalizedGlob = glob.replace(/\\/g, "/").replace(/^\.\//, "");
+        return normalizedTestPath.includes(normalizedGlob);
+      });
+    });
+  }
+
+  private async buildIgnoreGlobs(): Promise<string[]> {
+    const ignores = await vscode.workspace.findFiles("**/.gitignore", "**/node_modules/**");
+    const ignoreGlobs = ["**/node_modules/**"];
+
+    for (const ignore of ignores) {
+      try {
+        const content = await fs.readFile(ignore.fsPath, "utf8");
+        const lines = content
+          .split("\n")
+          .map(line => line.trim())
+          .filter(line => line && !line.startsWith("#"));
+
+        const cwd = path.relative(this.workspaceFolder.uri.fsPath, path.dirname(ignore.fsPath));
+
+        for (const line of lines) {
+          if (!cwd || cwd === "" || cwd === ".") {
+            ignoreGlobs.push(line.trim());
+          } else {
+            ignoreGlobs.push(path.join(cwd.trim(), line.trim()));
+          }
+        }
+      } catch (err) {
+        // Silent error handling
```

does vscode offer a way to report errors but not be immediately visible to the user? Not fond of quietly eating errors - someone could fall into this case and never know why the feature is broken

---

done!